app/src/main/java/com/pvp/app/ui/screen/settings/SettingsScreen.kt

```diff
@@ -190,13 +221,22 @@ fun CategoryRow(
 }
 
 @Composable
-fun SettingCard(
+fun <T> SettingCard(
     description: String,
-    editContent: @Composable () -> Unit,
+    editContent: @Composable () -> Unit = {},
     onEdit: () -> Unit,
-    title: String? = null,
-    value: String
+    title: String,
+    value: T,
+    isEnabled: Boolean = true
 ) {
+    var textColor = MaterialTheme.colorScheme.onPrimary
+    var backgroundColor = MaterialTheme.colorScheme.primary
+
+    if (!isEnabled) {
+        textColor = textColor.copy(alpha = 0.5f)
```

Comment:
```kotlin
.lighten()
.darken()
```
Our extensions exist that do exactly this and accept ratio in case you want to provide something else than 0.5